tests/Unit/Core/Foundation/VersionTest.php

```diff
@@ -0,0 +1,624 @@
+                    'majorString' => '1.7',
+                    'major' => 7,
+                    'minor' => 6,
+                    'patch' => 0,
+                    'preRelease' => '',
+                    'buildMeta' => '',
+                ],
+            ],
+            '7.6.0-dev' => [
+                '7.6.0-dev',
+                [
+                    'version' => '1.7.6.0',
+                    'fullVersion' => '1.7.6.0-dev',
+                    'semVersion' => '7.6.0-dev',
+                    'majorString' => '1.7',
+                    'major' => 7,
+                    'minor' => 6,
+                    'patch' => 0,
+                    'preRelease' => 'dev',
+                    'buildMeta' => '',
+                ],
+            ],
+            '7.6.0+test.build' => [
+                '7.6.0+test.build',
+                [
+                    'version' => '1.7.6.0',
+                    'fullVersion' => '1.7.6.0+test.build',
+                    'semVersion' => '7.6.0+test.build',
+                    'majorString' => '1.7',
+                    'major' => 7,
+                    'minor' => 6,
+                    'patch' => 0,
+                    'preRelease' => '',
+                    'buildMeta' => 'test.build',
+                ],
+            ],
+            '7.0-beta.1+build.156' => [
+                '7.7.0-beta.1+build.156',
+                [
+                    'version' => '1.7.7.0',
+                    'fullVersion' => '1.7.7.0-beta.1+build.156',
+                    'semVersion' => '7.7.0-beta.1+build.156',
+                    'majorString' => '1.7',
+                    'major' => 7,
+                    'minor' => 7,
+                    'patch' => 0,
+                    'preRelease' => 'beta.1',
+                    'buildMeta' => 'build.156',
+                ],
+            ],
+            '7.7.0-dev+nightly.20190526' => [
+                '7.7.0-dev+nightly.20190526',
+                [
+                    'version' => '1.7.7.0',
+                    'fullVersion' => '1.7.7.0-dev+nightly.20190526',
+                    'semVersion' => '7.7.0-dev+nightly.20190526',
+                    'majorString' => '1.7',
+                    'major' => 7,
+                    'minor' => 7,
+                    'patch' => 0,
+                    'preRelease' => 'dev',
+                    'buildMeta' => 'nightly.20190526',
+                ],
+            ],
+            '8.1.2' => [
+                '8.1.2',
+                [
+                    'version' => '1.8.1.2',
+                    'fullVersion' => '1.8.1.2',
+                    'semVersion' => '8.1.2',
+                    'majorString' => '1.8',
+                    'major' => 8,
+                    'minor' => 1,
+                    'patch' => 2,
+                    'preRelease' => '',
+                    'buildMeta' => '',
+                ],
+            ],
+        ];
+    }
+
+    public function getCompareGreater()
```

Comment:
Can you add tests for `compare` to make sure `version_compare` is working properly with `.7.7.0-dev+nightly.20190526`, `1.7.7.0-beta.1+build.156`, `1.7.6.0+test.build` for example.